docs/devel/networking.md

```diff
@@ -1,3 +1,84 @@
+is performed in the *target* virt-launcher's net namespace, as can be seen in
+the implementation of the [setPodNetworkPhase1 function](https://github.com/kubevirt/kubevirt/blob/8e7f2a411f5d06f1734e96c7bc08d0bb9ec1d500/pkg/virt-handler/vm.go#L340).
+
+This will in turn call the
+[PlugPhase1](https://github.com/kubevirt/kubevirt/blob/5995d2164faefdcc7a59a857e12f8ed4cdb7e094/pkg/virt-launcher/virtwrap/network/podinterface.go#L84),
+which performs the following operations:
+- `getPhase1Binding`: this function initializes the interface structures, and
+  selects the correct `driver` to use. The driver, is essentially a
+  `BindMechanism`. The next operations will all be of specific implementations
+  of [BindMechanisms](#binding-mechanisms).
+
+- `discoverPodNetworkInterface`: Each `BindMechanism` requires different
+   information about the pod interface - slirp, for instance, doesn't require
+   any info. The others, gather the following information:
+   - IP address
+   - Routes (**only** bridge)
+   - Gateway
+   - MAC address (**only** bridge)
+   - Link MTU
+
+- `preparePodNetworkInterfaces`: this function will make use of the
+  aforementioned information, performing actions specific to each `BindMechanism`.
+  See each `BindMechanism` section for more details.
+
+- `setCachedInterface`: caches the interface in memory
+
+- `setCachedVIF`: this will persist the
+  [VIF](https://github.com/kubevirt/kubevirt/blob/51ce9d8e6def33c0c260f0cdb994f08ae5b9938e/pkg/virt-launcher/virtwrap/network/common.go#L48)
+  object in the file system, making the configured pod interface available to
+  the virt-launcher pod. The VIF is cached in
+  `/proc/<virt-launcher-pid>/root/var/run/kubevirt-private/vif-cache-<iface_name>.json`.
+
+### Unprivileged VMI networking configuration
+The virt-launcher is an untrusted component of KubeVirt (since it wraps the
+libvirt process that will run third party workloads). As a result, it must be
+run with as little privileges as required. As of now, the only capability
+required by virt-launcher to configure networking is the `CAP_NET_ADMIN`
+capability.
+
+In this second phase, virt-launcher retrieves the configuration data previously
+gathered in phase #1 (loads the cached VIF object), and decorates the domain
+xml configuration of the VM it will encapsulate. This *decoration* is specific
+to each [binding mechanism](#binding-mechanisms), as each of those can involve
+different libvirt configurations.
+
+## Binding Mechanisms
+A binding mechanism can be seen as the translation service between KubeVirt's
+API and Libvirt's domain xml. Each interface type has a different binding
+mechanism, since it will lead to a different libvirt domain xml specification.
+
+As of now, the existent binding mechanisms are:
+- [bridge](#bridge-binding-mechanism)
+- [masquerade](#masquerade-binding-mechanism)
+- [slirp](#slirp-binding-mechanism)
+
+### Bridge binding mechanism
```

Comment:
if we aren't going to fill in these sections in this pr, can we add TODOs here?

Reply:
Done.

Reply:
I'm trying to squeeze in a discussion about the structure about the document's structure into this PR.

If you prefer, I can take this off, and do that on a separate PR. Or, if we agree this structure is OK, I can simply remove this.